test/integration/tls/tls_stack.go

```diff
@@ -0,0 +1,908 @@
+				f.Logger.Error("failed to delete namespace",
+					zap.String("namespace", s.namespace.Name),
+					zap.Error(err))
+				deletionErrors = append(deletionErrors, err)
+				return
+			}
+
+			By(fmt.Sprintf("wait namespace to be deleted: %s", s.namespace.Namespace), func() {
+				if err := f.NSManager.WaitUntilNamespaceDeleted(ctx, s.namespace); err != nil {
+					f.Logger.Error("failed to wait namespace deletion",
+						zap.String("namespace", s.namespace.Name),
+						zap.Error(err))
+					deletionErrors = append(deletionErrors, err)
+				}
+			})
+		})
+	}
+
+	if s.mesh != nil {
+		By(fmt.Sprintf("delete mesh %s", s.mesh.Name), func() {
+			if err := f.K8sClient.Delete(ctx, s.mesh,
+				client.PropagationPolicy(metav1.DeletePropagationForeground), client.GracePeriodSeconds(0)); err != nil {
+				f.Logger.Error("failed to delete mesh",
+					zap.String("mesh", s.mesh.Name),
+					zap.Error(err))
+				deletionErrors = append(deletionErrors, err)
+				return
+			}
+
+			By(fmt.Sprintf("wait mesh to be deleted: %s", s.mesh.Name), func() {
+				if err := f.MeshManager.WaitUntilMeshDeleted(ctx, s.mesh); err != nil {
+					f.Logger.Error("failed to wait mesh deletion",
+						zap.String("mesh", s.mesh.Name),
+						zap.Error(err))
+					deletionErrors = append(deletionErrors, err)
+				}
+			})
+		})
+	}
+
+	return deletionErrors
+}
+
+func (s *TLSStack) createSecretsForTLSStack(ctx context.Context, f *framework.Framework, mb *manifest.ManifestBuilder) {
+	By("create secrets to be used by frontend for validation", func() {
+		frontendTLSFiles := []string{"ca_1_cert.pem"}
+		secret := mb.BuildK8SSecretsFromPemFile("tls_app/tls/", frontendTLSFiles,
+			"ca1-cert-tls", f)
+		err := f.K8sClient.Create(ctx, secret)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	By("create secrets to be used by backend", func() {
+		backendTLSFiles := []string{"backend-tls_cert_chain.pem", "backend-tls_key.pem"}
+		secret := mb.BuildK8SSecretsFromPemFile("tls_app/tls/", backendTLSFiles,
+			"backend-tls-tls", f)
+		err := f.K8sClient.Create(ctx, secret)
+		Expect(err).NotTo(HaveOccurred())
+	})
+}
+
+func (s *TLSStack) createSecretsForPartialTLSStack(ctx context.Context, f *framework.Framework, mb *manifest.ManifestBuilder) {
+	By("create secrets to be used by frontend for validation", func() {
+		frontendTLSFiles := []string{"ca_1_cert.pem"}
+		secret := mb.BuildK8SSecretsFromPemFile("tls_app/tls/", frontendTLSFiles,
+			"ca1-cert-tls", f)
+		f.Logger.Error("Secret: ", zap.String("Name: ", secret.Name), zap.String("Namespace: ", secret.Namespace))
+		err := f.K8sClient.Create(ctx, secret)
+		Expect(err).NotTo(HaveOccurred())
+	})
+}
+
+func (s *TLSStack) createVirtualNodeResourcesForTLSStack(ctx context.Context, f *framework.Framework, mb *manifest.ManifestBuilder) {
+	By("create virtualNode resources", func() {
+		vnBuilder := &manifest.VNBuilder{
+			ServiceDiscoveryType: manifest.DNSServiceDiscovery,
```

Comment:
Should be `ServiceDiscoveryType: s.ServiceDiscoveryType` to avoid overriding the value passed down from the test

Reply:
Not setting up anything at Stack level right now.